### PR TITLE
Add : dynamic wallet logo

### PR DESCRIPTION
**Pull request by shadabkhan16**
## Description
The ``` useWallet ``` hook is designed to determine the currently authenticated wallet based on the user's authentication state, map the wallet type to a corresponding image path, and return the image path for the wallet.

- What does this PR do?
- Why is it needed?

## Screenshots

Screen
shot 2024-12-09 at 5 25 51 PM
Screenshot 2024-12-09 at 5 25 44 PM

### Desktop:
[Add screenshot here]

### Mobile:
[Add screenshot here]

## Approach
Briefly explain the approach taken to implement the changes. 

- How did you solve the problem?
- Any particular design patterns or optimizations?

## Issue Number (if any)
Link to the issue this PR addresses, if applicable:

Fixes #[issue_number]

## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/game_frontend/src/hooks/useWallet.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { useAuth } from "./useAuth";
+
+const useWallet = (): string | undefined => {
+  const auth = useAuth();
+
+
+
+  // Determine the current authenticated wallet
+  const currentWallet: WalletName | undefined = (() => {
```
Comment (ArjunQBTech) [inline src/game_frontend/src/hooks/useWallet.tsx:9]: Look into `useMemo` or `useCallback` and see which one will be more optimized solution for this function

File: src/game_frontend/src/hooks/useWallet.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  // Map wallet types to image paths
+  const WALLETS: Record<WalletName, string> = {
+    PLUG: "/assets/images/wallets/plug.png",
+    INTERNETIDENTITY: "/assets/images/wallets/ii.png",
+    BIFINITY: "/assets/images/wallets/bifinity.png",
+    NFID: "/assets/images/wallets/nfid.png",
+  };
+
+  // Return the wallet image path
+  const walletImage = currentWallet ? WALLETS[currentWallet] : undefined;
```
Comment (ArjunQBTech) [inline src/game_frontend/src/hooks/useWallet.tsx:24]:
```js
const walletImage = currentWallet ? WALLETS[currentWallet] : '/assets/images/do-not-disturb.svg';
```

File: src/game_frontend/src/components/ui/Navbar/index.tsx [hunk 3]
```diff
@@ -1,14 +1,16 @@
 import React from "react";
 import "./index.css";
+import useWallet from "../../../hooks/useWallet";
 
 const Navbar: React.FC<NavbarProps> = ({ profitPerHour = 0.3 }) => {
+  const WalletImage = useWallet() || "/assets/images/do-not-disturb.svg";
```
Comment (ArjunQBTech) [inline src/game_frontend/src/components/ui/Navbar/index.tsx:6]: Remove fallback image from here and move to the hook itself

**Review by ArjunQBTech — CHANGES_REQUESTED** [review]

File: src/game_frontend/src/hooks/useWallet.tsx (hunk 1 above)
Comment (shadabkhan16) [reply src/game_frontend/src/hooks/useWallet.tsx]: Going with ```useMemo```

**Review by ArjunQBTech — APPROVED** [review]
Great! Looks good

**Pull request merged**